src/shared/fdset.c

```diff
@@ -246,10 +262,9 @@ int fdset_new_listen_fds(FDSet **ret, bool unset) {
         return 0;
 }
 
-int fdset_to_array(FDSet *fds, int **ret) {
-        unsigned j = 0, m;
-        void *e;
-        int *a;
+static int to_array(FDSet *fds, bool indexed, int **ret) {
```

Comment:
hmm, we don't allow mixing indexed and non-indexed entries in the same fdset, do we?

hence maybe add a helper fdset_is_indexed() that checks whether the first entry (via hashmap_first()) is indexed or not, and then use that instead of an explicit `indexed` parameter).

and in fact, maybe refuse adding non-indexed insertions if fdset_is_indexed() returns true, and vice versa

Reply:
is looking at the first element enough?

Reply:
if we don't allow mixing both types of entries it should be enough

Reply:
mmh no doesn't seem to work, I have an fdset with 25, 26 and 27 at indexes 0, 1 and 2 respectively, and hashmap_first is returning me 27

Reply:
I need to check both hashmap_first and then use it again to see if it is a key itself again to make it work